lib/fastlane/actions/deliver.rb

```diff
@@ -11,8 +11,8 @@ def self.run(config)
           FastlaneCore::UpdateChecker.start_looking_for_update('deliver') unless Helper.is_test?
 
           config.load_configuration_file("Deliverfile")
-          config[:screenshots_path] ||= Actions.lane_context[SharedValues::SNAPSHOT_SCREENSHOTS_PATH] # use snapshot's screenshots
-          config[:ipa] ||= Actions.lane_context[SharedValues::IPA_OUTPUT_PATH]
+          config[:screenshots_path] = Actions.lane_context[SharedValues::SNAPSHOT_SCREENSHOTS_PATH] if Actions.lane_context[SharedValues::SNAPSHOT_SCREENSHOTS_PATH]
+          config[:ipa] = Actions.lane_context[SharedValues::IPA_OUTPUT_PATH] if Actions.lane_context[SharedValues::IPA_OUTPUT_PATH]
```

Comment:
A bit nitpicky, but because those context expressions are so unwieldy, I'd prefer:

```ruby
context_screenshots_path = Actions.lane_context[SharedValues::SNAPSHOT_SCREENSHOTS_PATH]
config[:screenshots_path] = context_screenshots_path if context_screenshots_path

context_ipa_output_path = Actions.lane_context[SharedValues::IPA_OUTPUT_PATH]
config[:ipa] = context_ipa_output_path if context_ipa_output_path
```